### PR TITLE
Locale supported as timezone

### LINKED ISSUE #49: Bug report: Timezone offset potential issue
Please check these things before submitting your issue:

- [x] I am using the latest version
- [x] I searched for duplicate or closed bug report [here](https://github.com/PuneetGopinath/github-activity-readme/issues?q=is%3Aissue)

**Describe the bug**
I love the new changes with last update and timezone details, but I believe there will be an issue here related to this. For example, there are countries, or maybe just Spain, who's timezone changes during daylight savings time. During DST, the majority of Spain uses +2 for the offset, but outside of DST, it's +1 as the offset. I have the same issue with my DSLR camera where I have to change this twice a year because it uses offsets rather than just selecting a region/capital. I am not sure if this is something that can be fixed or if it is considered a bug, but I thought it is worth mentioning. Perhaps the best solution is just to keep it neutral at GTM 0.

**To Reproduce**
It's time based so maybe make a run during DST and outside of DST changing the local date as per references below.

**Expected behavior**
It would be nice to be able to set by region|capital and have the library workout the correct offset.

**Screenshots**
N/A

**Additional context**
You can see references for both of these shifting timezones here:
https://www.timeanddate.com/time/zone/spain/madrid
https://en.wikipedia.org/wiki/Time_in_Spain#IANA_time_zone_database

### PR DESCRIPTION
**Pull request by abhijoshi2k**
## Checks

I have...

- [x] read and understood the [Contributing Guidelines][contributing]
- [x] Updated README.md.
- [ ] Updated CHANGELOG.md.

## Type of Pull request

- [ ] **Minor Change**  
  This Pull request doesn't break existing configuration.
- [x] **Major Change**  
  This Pull request will break existing configuration.
- [x] **Bug fix**  
  This Pull request will fix a (critical) bug.
- [ ] **Documentation**  
  This Pull request only changes documentation (README.md, CHANGELOG.md, etc.)

## Description

Timezone support extended. It can now accept locale string as "continent/city"
f.e. GMT+0530 or Asia/Kolkata
Formats for all timezones can be viewed here : https://en.wikipedia.org/wiki/List_of_tz_database_time_zones

### 🔴🔴 Breaking Change 🔴🔴
`TIMEZONE_OFFSET` input has now changed to `TIMEZONE`

Closes #49 

[contributing]: https://github.com/Readme-Workflows/recent-activity/blob/main/.github/CONTRIBUTING.md

### CODE REVIEW

**Comment by abhijoshi2k** [comment]
@Andre601 We'll be good to resolve #49 after you update readme and changelog.

**Comment by PuneetGopinath** [comment]
I test the branch in https://github.com/PuneetGopinath/activity-test nowm

**Comment by PuneetGopinath** [comment]
@abhijoshi2k See those results in https://github.com/PuneetGopinath/activity-test/actions/runs/866856864
They passed

**Comment by PuneetGopinath** [comment]
@abhijoshi2k Is there something you want to still commit in the timezone-fix branch?

**Comment by abhijoshi2k** [comment]
> @abhijoshi2k Is there something you want to still commit in the timezone-fix branch?

Changelog is still left

**Comment by abhijoshi2k** [comment]
@PuneetGopinath let us stop all the work on #53 till this pr is merged and all conflicts are resolved.

**Comment by PuneetGopinath** [comment]
@abhijoshi2k Why do we have to release minor version can't we release a new major version, v1.3.0 of our release has not followed semantic version already, can I update the Changelog and the draft release?

**Comment by abhijoshi2k** [comment]
> @abhijoshi2k Why do we have to release minor version can't we release a new major version, v1.3.0 of our release has not followed semantic version already, can I update the Changelog and the draft release?

Yes, we can go for v2.0.0 in next release

**Comment by Andre601** [comment]
That's not how semantic versioning works imo

**Comment by PuneetGopinath** [comment]
> That's not how semantic versioning works imo.
> 1.4.0 is perfectly fine...

How?

**Comment by PuneetGopinath** [comment]
Semver says to use a new major version when there is a breaking change.

**Comment by abhijoshi2k** [comment]
> That's not how semantic versioning works imo.
> 1.4.0 is perfectly fine...

Let's release new version after #53 is merged.
It has totally new structure and major breaking changes. I think it'll be worth v2.0

**Review by Andre601 — APPROVED** [review]

**Review by Andre601 — CHANGES_REQUESTED** [review]
I dislike that we delete the old changelog.
This goes completely against the purpose of a Change**LOG**

File: functions/appendDate.js [hunk 1]
```diff
@@ -18,23 +18,29 @@ const appendDate = (fullContent) => {
...
-    let timezone = TIMEZONE_OFFSET.replace("GMT", "").split(":");
     let offset;
+    let finalDate;
 
-    let tz_hours = parseInt(timezone[0].trim());
-
-    if (timezone.length > 1) {
-      offset = tz_hours * 60 + parseInt(timezone[1].trim());
+    if (TIMEZONE.split("/").length === 2) {
+      process.env.TZ = TIMEZONE;
```
Comment (PuneetGopinath) [inline functions/appendDate.js:25]: @abhijoshi2k Sorry, I don't know from where the `process.env.TZ` came, is the process variable built-in in node.js

**Comment by abhijoshi2k** [comment]
> I dislike that we delete the old changelog.
> This goes completely against the purpose of a Change**LOG**

@PuneetGopinath I agree with @Andre601 Let's not delete the old changelog.

File: functions/appendDate.js (hunk 1 above)
Comment (abhijoshi2k) [reply functions/appendDate.js:25]: Yes it's built in

**Comment by PuneetGopinath** [comment]
> I dislike that we delete the old changelog.
> This goes completely against the purpose of a Change**LOG**

No, we should reset it when a new version is released, that's why in the old format, I added the older changelogs section you removed it!!

**Comment by abhijoshi2k** [comment]
@PuneetGopinath @Andre601  can we keep changelogs of old version in separate folder then?

**Comment by PuneetGopinath** [comment]
This doesn't

> @PuneetGopinath @Andre601 can we keep changelogs of old version in separate folder then?

Yes, we can keep them, anyway they are available in the v1.3.0 tag

**Comment by Andre601** [comment]
> @PuneetGopinath @Andre601 can we keep changelogs of old version in separate folder then?

At least do that and link to it in the current change log...
But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.

**Comment by PuneetGopinath** [comment]
> At least do that and link to it in the current change log...
> But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.

Don't become angery 😆

**Comment by PuneetGopinath** [comment]
Can I merge.

**Comment by abhijoshi2k** [comment]
> At least do that and link to it in the current change log...
> But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.

There's no need to get angry @Andre601 .
Let's work as a team. I don't think getting angry and exiting is a solution here. Mistakes do happen. Let's be considerate. 😊

**Comment by abhijoshi2k** [comment]
> Can I merge.

Let's not hurry. Wait for @Andre601 to respond.

**Comment by PuneetGopinath** [comment]
@Andre601 Only one chance went, there are 99+ more chances left!! Never become angry!

**Comment by Andre601** [comment]
> > At least do that and link to it in the current change log...
> > But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.
> 
> Don't become angery 😆

I am.
I work more than enough with projects who use semantic versioning and now you come here and essentially piss on this.
I never in my life saw any project that jumps to a new major version for some feature changes. So far I only saw actual major changes happening on entire recodes of the code. Everything else at most caused a minor change.

The actual, true semantic versioning is `major.minor.patch_build` where major is for recodes/extreme code changes, minor for changes without backwards compatibility, patch for changes with backwards compatibility and build for bug fixes and similar.

See for example [JDA](https://github.com/DV8FromTheWorld/JDA)

It's just annoying that I'm here for README and CHANGELOG changes, yet you two take this over now and I get the back seat and reviewer role...

**Comment by PuneetGopinath** [comment]
I hurried because, am going to sleep

**Comment by Andre601** [comment]
Just make a backup of the old (v1) changelog now... Maybe in a `.github/archives` folder and named `CHANGELOG_v1.md`
That's all I ask for now.

**Comment by abhijoshi2k** [comment]
> I hurried because, am going to sleep

I shall merge it after we clear everything with @Andre601 . I don't want to hurt a team member here. I shall also resolve all the conflicts. You can sleep.

**Comment by PuneetGopinath** [comment]
> Just make a backup of the old (v1) changelog now... Maybe in a `.github/archives` folder and named `CHANGELOG_v1.md`
> That's all I ask for now.

@abhijoshi2k You can get the old change log from the old changelogs section, then copy the raw content and create a new file.

**Comment by abhijoshi2k** [comment]
> > > At least do that and link to it in the current change log...
> > > But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.
> > 
> > Don't become angery 😆
> 
> I am.
> I work more than enough with projects who use semantic versioning and now you come here and essentially piss on this.
> I never in my life saw any project that jumps to a new major version for some feature changes. So far I only saw actual major changes happening on entire recodes of the code. Everything else at most caused a minor change.
> 
> The actual, true semantic versioning is `major.minor.patch_build` where major is for recodes/extreme code changes, minor for changes without backwards compatibility, patch for changes with backwards compatibility and build for bug fixes and similar.
> 
> See for example [JDA](https://github.com/DV8FromTheWorld/JDA)
> 
> It's just annoying that I'm here for README and CHANGELOG changes, yet you two take this over now and I get the back seat and reviewer role...

@PuneetGopinath let's make sure this doesn't happen anymore.
@Andre601 I'm sorry that this happened.
Let's put it in past and start fresh.
I hope you understand.

**Comment by PuneetGopinath** [comment]
@abhijoshi2k We have breaking changes and that should release a new major version

**Comment by Andre601** [comment]
I merged a `.github/changelogs/CHANGELOG_v1.md` file into this branch now.

**Review by Andre601 — APPROVED** [review]

**Comment by StevenJDH** [comment]
> 
> 
> > > At least do that and link to it in the current change log...
> > > But whatever. I don't have anything to do here anymore, since you two took over my only purpose here. So do wtf you want Idc anymore.
> > 
> > 
> > Don't become angery 😆
> 
> I am.
> I work more than enough with projects who use semantic versioning and now you come here and essentially piss on this.
> I never in my life saw any project that jumps to a new major version for some feature changes. So far I only saw actual major changes happening on entire recodes of the code. Everything else at most caused a minor change.
> 
> The actual, true semantic versioning is `major.minor.patch_build` where major is for recodes/extreme code changes, minor for changes without backwards compatibility, patch for changes with backwards compatibility and build for bug fixes and similar.
> 
> See for example [JDA](https://github.com/DV8FromTheWorld/JDA)
> 
> It's just annoying that I'm here for README and CHANGELOG changes, yet you two take this over now and I get the back seat and reviewer role...

Hello @Andre601 and all, I just wanted to provide my two cents since my original ticket spurred this discussion and I don't want to see people upset on great working being done already. Debating can be a good thing and even healthy as long as no one takes it personal, and with versioning, there are always a lot of opinions. With that said, I will just share this, https://semver.org/. These are very clear guidelines, and with regards to Minor version, it says `MINOR version when you add functionality in a backwards compatible manner.` The key here is minor is not for breaking changes, but backwards compatible ones such as new features, or perhaps supporting new breaking changes of existing features while still supporting the old way marked as deprecated.

Keeping in mind the above, I think, at least in part, that some of the confusion is coming from releasing maybe too few breaking changes as a new version. For example, 1 breaking change causing the major version to update each time, which feels wrong as opposed to maybe 10. The discussion above seems to hover around both of these concepts. I won't say what I think is best, but I did just want to share some info in the hopes it helps work things out easier.

Thanks again everyone for your time and involvement in this.

**Comment by abhijoshi2k** [comment]
Hello @StevenJDH ,
Thanks for the valuable information provided by you. We will surely go through it and take proper steps.
I hope that issue you faced regarding DST has been resolved.
Also, I wish to let you know that if you are using the `main` branch for your workflow, please note that the file structure is going to be changed after #53 will be merged. Surely, the readme will be updated accordingly.
We will make sure we mention you in here to remind whenever #53 is merged and new version is released.

Thank you.

**Comment by StevenJDH** [comment]
@abhijoshi2k yes, things look good. And yes, I've decided to switch from main to a version number after reconsidering my approach here. Thanks again 🙏🏽

**Comment by abhijoshi2k** [comment]
@StevenJDH The feature you requested is now released with v2.0.0
You can now use it with separate config file. Documentation is available [here](https://github.com/Readme-Workflows/recent-activity/wiki)

**Comment by StevenJDH** [comment]
@abhijoshi2k perfect, I will upgrade today, thanks!

**Pull request merged**